### PR TITLE
add Colorbit

### PR DESCRIPTION
**Pull request by microbit-mark**
add Colorbit extension @xuefengedu

### CODE REVIEW

**Comment by xuefengedu** [comment]
seems a travis issue:
```
$ node node_modules/pxt-core/built/pxt.js travis
Using target microbit with build engine yotta
  target: v /home/travis/build/Microsoft/pxt-microbit
  pxt-core: v /home/travis/build/Microsoft/pxt-microbit/node_modules/pxt-core
TRAVIS_TAG: 
TRAVIS_BRANCH: update-extensions
TRAVIS_PULL_REQUEST: false
uploadLocs: false
latest: git-update-extensions
target build
log strings: 34 files; 0 strings -> sim-strings.json
copying common-sim...
error TS6053: File '/home/travis/build/Microsoft/pxt-microbit/node_modules/pxt-core/built/lib.dom.d.ts' not found.
error TS6053: File '/home/travis/build/Microsoft/pxt-microbit/node_modules/pxt-core/built/lib.dom.iterable.d.ts' not found.
error TS6053: File '/home/travis/build/Microsoft/pxt-microbit/node_modules/pxt-core/built/lib.es2015.d.ts' not found.
error TS6053: File '/home/travis/build/Microsoft/pxt-microbit/node_modules/pxt-core/built/lib.scripthost.d.ts' not found.
error TS2318: Cannot find global type 'Array'.
error TS2318: Cannot find global type 'Boolean'.
error TS2318: Cannot find global type 'Function'.
error TS2318: Cannot find global type 'IArguments'.
error TS2318: Cannot find global type 'Number'.
error TS2318: Cannot find global type 'Object'.
error TS2318: Cannot find global type 'RegExp'.
error TS2318: Cannot find global type 'String'.
building target.json in /home/travis/build/Microsoft/pxt-microbit...
building /home/travis/build/Microsoft/pxt-microbit/libs/core
test.ts(9,1): error TS2304: Cannot find name 'forever'.
test.ts(10,5): error TS2304: Cannot find name 'pause'.
test.ts(9,1): error TS9235: Unknown or undeclared identifier
test.ts(9,1): error TS9201: unsupported type: any 0x1
test.ts(10,5): error TS9235: Unknown or undeclared identifier
test.ts(10,5): error TS9201: unsupported type: any 0x1
package built; written to binary.hex
building /home/travis/build/Microsoft/pxt-microbit/libs/radio
package built; written to binary.hex
building /home/travis/build/Microsoft/pxt-microbit/libs/devices
package built; written to binary.hex
building /home/travis/build/Microsoft/pxt-microbit/libs/bluetooth
package built; written to binary.hex
building /home/travis/build/Microsoft/pxt-microbit/libs/servo
package built; written to binary.hex
building /home/travis/build/Microsoft/pxt-microbit/libs/radio-broadcast
package built; written to binary.hex
building libs/core
building libs/radio
building libs/devices
building libs/bluetooth
building libs/servo
building libs/radio-broadcast
building libs/blocksprj
created native image in offline cache for project libs/blocksprj: /home/travis/build/Microsoft/pxt-microbit/built/hexcache/13fc432659dae036c336a1793eb016022d77abca65ae39c86dc5d1b2314eb10d.hex
building libs/bluetoothprj
created native image in offline cache for project libs/bluetoothprj: /home/travis/build/Microsoft/pxt-microbit/built/hexcache/36ef2616d2f5c491573ffc33d72099c643df15b9cd38d6bfceec738b3948d960.hex
building libs/tsprj
native image already in offline cache for project libs/tsprj: /home/travis/build/Microsoft/pxt-microbit/built/hexcache/13fc432659dae036c336a1793eb016022d77abca65ae39c86dc5d1b2314eb10d.hex
INTERNAL ERROR: SyntaxError: Unexpected token ] in JSON at position 4570
    at JSON.parse (<anonymous>)
    at Object.readJson (/home/travis/build/Microsoft/pxt-microbit/node_modules/pxt-core/built/nodeutil.js:245:17)
    at saveThemeJson (/home/travis/build/Microsoft/pxt-microbit/node_modules/pxt-core/built/pxt.js:132957:39)
    at /home/travis/build/Microsoft/pxt-microbit/node_modules/pxt-core/built/pxt.js:133258:9
    at tryCatcher (/home/travis/build/Microsoft/pxt-microbit/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/travis/build/Microsoft/pxt-microbit/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/home/travis/build/Microsoft/pxt-microbit/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/home/travis/build/Microsoft/pxt-microbit/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/home/travis/build/Microsoft/pxt-microbit/node_modules/bluebird/js/release/promise.js:693:18)
    at Async._drainQueue (/home/travis/build/Microsoft/pxt-microbit/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/home/travis/build/Microsoft/pxt-microbit/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/home/travis/build/Microsoft/pxt-microbit/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:789:20)
    at tryOnImmediate (timers.js:751:5)
    at processImmediate [as _immediateCallback] (timers.js:722:5)
The command "node node_modules/pxt-core/built/pxt.js travis" exited with 20.
cache.2
store build cache
0.00s1.74snothing changed, not updating cache
Done. Your build exited with 1.
```

**Comment by xuefengedu** [comment]
@microbit-mark @pelikhan seems travis issue, can you help to re-trigger travis job? thanks very much!

**Comment by xuefengedu** [comment]
@microbit-mark is it because you are using `"51bit/ColorBit",` instead of `"51bit/ColorBit"` in targetconfig.json, mine is `"51bit/ColorBit"`,  thanks.

File: targetconfig.json [hunk 1]
```diff
@@ -102,7 +102,8 @@
             "PiSupply/pxt-oled-ssd1306",
             "pimoroni/pxt-touchbit",
             "4tronix/cubebit",
-            "4tronix/BitCommander"
+            "4tronix/BitCommander",
+            "51bit/ColorBit",
```
Comment (xuefengedu) [inline targetconfig.json]: this seems should be `"51bit/ColorBit"`
not `"51bit/ColorBit",`
thanks for help!

Comment (microbit-mark) [reply targetconfig.json]: yup. my bad. Done

**Comment by xuefengedu** [comment]
@microbit-mark passed now, thanks for great help!

**Comment by xuefengedu** [comment]
@pelikhan could you help to review and merge this PR? thanks very much for great help!

**Comment by pelikhan** [comment]
@microbit-mark conflicts

**Comment by xuefengedu** [comment]
thanks everyone's help on this PR!

**Comment by pelikhan** [comment]
Please fix conflicts

**Comment by xuefengedu** [comment]
@microbit-mark update-extension branch need to rebase from master, you might use these commands or just edit the conflicts on this page when merging PR.
```
git checkout update-extensions
git rebase master
if shows conflicts, just use vim or notepad++ to edit the conflict file, there are some >>> as begin, == as split, << as end, just merge the begin part and the end part into one, then remove all symbols:<<==>>, then save and exit
once modify all conflict files, you can run git add <your modified files' file path +name>; git rebase --continue; 
last step: git push origin update-extensions -f
then there will not have conflicts
```

**Comment by xuefengedu** [comment]
don't forget to sync master branch from github before rebase if you want command line way, thanks.

```
git fetch
git checkout master
git pull origin master
git reset --hard HEAD
```

**Comment by xuefengedu** [comment]
if possible, I can help your team doing some   work in the future, I mean possible be a committer in the future? :D

**Comment by pelikhan** [comment]
We blandly take Pull Requests!

**Comment by xuefengedu** [comment]
@pelikhan Got you, thanks for reviewing!

**Comment by xuefengedu** [comment]
@microbit-mark can you help to fix the conflicts with master branch? thanks very much for help!

**Comment by xuefengedu** [comment]
@pelikhan in fact, when merge PR on this page, it will show "Resolve Conflicts" button which can be clicked to resolve the conflicts files.
there are some >>> as begin, == as split, << as end, just merge the begin part and the end part into one, then remove all symbols:<<==>>, then click "Mark as resolved" for that file.
after mark the 2 conflicts files, it can be merged into master, thanks very much!

I don't have permission to resolve it, thanks.

**Comment by xuefengedu** [comment]
@microbit-mark passed with no conflicts now, thanks so much!

**Comment by xuefengedu** [comment]
@pelikhan no conflicts now, can you help to merge it to master? thanks very much!

**Comment by xuefengedu** [comment]
seems some problems:

```
  "name": "51bit ColorBit",
	  "url":"/pkg/51bit/ColorBit",
	  "name": "MAX7219 8x8",
	  "url":"/pkg/alankrantas/pxt-MAX7219_8x8",
	  "cardType": "package
```

File: docs/extensions.md [hunk 2]
```diff
@@ -313,6 +313,8 @@ Check out [microbit.org](http://microbit.org/resellers/) for more information on
   "url":"/pkg/alsrobot-microbit-makecode-packages/ALSRobotJoyBit",
   "cardType": "package"
 }, {
+  "name": "51bit ColorBit",
+  "url":"/pkg/51bit/ColorBit",
   "name": "MAX7219 8x8",
```
Comment (xuefengedu) [inline docs/extensions.md]: 2 name in one `{}`, seems incorrect.

Comment (xuefengedu) [reply docs/extensions.md]: @microbit-mark you can manually modify the `docs/extensions.md` file ,then add and push a new commit based on current situation.

Comment (microbit-mark) [reply docs/extensions.md]: fixed

**Comment by xuefengedu** [comment]
Thanks @microbit-mark  for help updating!

**Comment by xuefengedu** [comment]
@pelikhan no conflicts and no issue now, can you help to merge it to master? thanks very much!

**Comment by xuefengedu** [comment]
thanks everyone's help on this PR, thanks so much!

**Pull request merged**